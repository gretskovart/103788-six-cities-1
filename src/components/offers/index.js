import Offers from './offers.jsx';

export default Offers;
