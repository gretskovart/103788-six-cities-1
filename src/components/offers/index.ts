// @ts-ignore
import Offers from './offers.tsx';

export default Offers;
