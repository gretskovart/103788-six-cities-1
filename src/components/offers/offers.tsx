import * as React from 'react';
import {connect} from 'react-redux';
import PlacesList from '../places-list';
import PlacesFound from '../places-found';
import Map from '../map';
import Sorting from '../sorting';
import {types} from '../../helpers';

interface Props {
  offers: types.OfferType[];
}

const Offers = (props: Props) => {
  const {offers} = props;

  return (
    <div className="cities__places-container container">
      <section className="cities__places places">
        <h2 className="visually-hidden">Places</h2>
        <PlacesFound offers={offers} />
        <Sorting />
        <div className="cities__places-list places__list tabs__content">
          <PlacesList offers={offers} />
        </div>
      </section>
      <div className="cities__right-section">
        <Map offers={offers} />
      </div>
    </div>
  );
};

const mapStateToProps = (state: {data: types.OfferType}) => {
  return {
    offers: state.data.offers
  };
};

export {Offers};

export default connect(
    mapStateToProps,
    null
)(Offers);
