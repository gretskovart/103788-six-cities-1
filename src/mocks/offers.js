export default [
  {
    id: `apartment-01`,
    img: `apartment-01.jpg`,
    isPremium: true,
    price: 120,
    rating: 93,
    title: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`
  },
  {
    id: `room`,
    img: `room.jpg`,
    isPremium: false,
    price: 80,
    rating: 80,
    title: `Wood and stone place`,
    type: `Private room`
  },
  {
    id: `apartment-02`,
    img: `apartment-02.jpg`,
    isPremium: false,
    price: 132,
    rating: 70,
    title: `Canal View Prinsengracht`,
    type: `Apartment`
  },
  {
    id: `apartment-03`,
    img: `apartment-03.jpg`,
    isPremium: true,
    price: 180,
    rating: 100,
    title: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`
  }
];
