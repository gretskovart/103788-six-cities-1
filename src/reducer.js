import offers from './mocks/offers';

const getPlaces = (selectedCity = `Amsterdam`) => offers.filter((it) => it.city === selectedCity);

const initialState = {
  activeCity: `Amsterdam`,
  offers: getPlaces()
};

const actionCreators = {
  changeCity: (selectedCity) => {
    return ({
      type: `changeCity`,
      payload: selectedCity
    });
  },
  getOffers: (selectedCity) => {
    const places = getPlaces(selectedCity);

    return ({
      type: `getOffers`,
      payload: places
    });
  }
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case `changeCity`:
      return Object.assign({}, state, {
        activeCity: action.payload,
      });
    case `getOffers`:
      return Object.assign({}, state, ({
        offers: action.payload
      }));
  }

  return state;
};

export {actionCreators, reducer};
