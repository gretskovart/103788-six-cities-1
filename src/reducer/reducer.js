import {actionCreators, reducer, getData} from './data/reducer';

export {actionCreators, reducer, getData};
