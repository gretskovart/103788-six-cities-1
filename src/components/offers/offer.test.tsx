import * as React from 'react';
import {Offers} from './offers';
import Enzyme, {mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {createStore} from 'redux';
import {Provider} from 'react-redux';
import reducer from '../../reducer';

const store = createStore(reducer);

Enzyme.configure({adapter: new Adapter()});

it(`<Offers /> renders correctly`, () => {
  const div = document.createElement(`div`);
  document.body.appendChild(div);

  const wrapper = mount(
      <Provider store={store}>
        <Offers
          offers={[]}
        />
      </Provider>
      , {attachTo: div});

  expect(wrapper).toMatchSnapshot();
});
