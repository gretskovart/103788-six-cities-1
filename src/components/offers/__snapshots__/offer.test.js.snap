// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Offers /> renders correctly 1`] = `ReactWrapper {}`;
